tfimm/train/trainer.py

```diff
@@ -172,7 +173,8 @@ def train(self):
 
                 it += 1
 
-            self.validation()
+            if self.val_ds:
```

Comment:
I would move this check into `self.validation()`. That way you need to check whether `val_ds is None` in only one place.